### PR TITLE
Add the signup flow

### PR DESCRIPTION
**Pull request by vsaarinen**
This adds a new route, `/signup/<teamToken>`, that redirect the user to Auth0 in order to create a new user. Once created, they are redirected back to `/signup?queryparams`, which triggers a signup attempt with the backend. If successful, the signup page will show the user their Git username and password details, after which they can continue on to Minard.

Note that this PR does not include any styling for the signup views.

To test locally, you will need to be able to create a team token. In order to do this, your Minard user needs to a) be in a GitLab group called "lucify" (i.e. the admin group) and b) have its username be the same as the Auth0 sub claim value (e.g. `auth0|djahsgkahjsflakj`). You can find instructions on how to do the second part in the [previous PR](https://github.com/lucified/minard-ui/pull/192).

Once these have been done, use the [team token generation script](https://github.com/lucified/minard-team-token) to generate a token for the desired team. After this, use `http://localtest.me:3000/signup/<token>` to sign up as a new user. If you use `http://localhost:3000`, it will work, but you will get an additional permission dialog when signing up in Auth0.

### CODE REVIEW

**Comment by vsaarinen** [comment]
Note: currently, signing up fails after you are redirected back to Minard from Auth0. This is because Auth0 is, for some reason, not including the team token in the JWT access token. If you keep your browser open (i.e. stay signed in to Auth0) and reopen the original `http://localtest.me:3000/signup/<teamtoken>` link, the signup should work.

**Comment by dennari** [comment]
Ah, only now noticed your comment about the access_token. Got it working by retrying the signup url. 

We should try if the method suggested by the Auth0 support works.

File: tsconfig.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "noUnusedParameters": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": "./types",
-    "paths": {}
+    "paths": {},
+    "skipLibCheck": true // TODO: remove once https://github.com/DefinitelyTyped/DefinitelyTyped/issues/14324 resolved
```
Comment (vsaarinen) [inline tsconfig.json]: Fixed this in an upcoming PR where I update packages.

**Comment by vsaarinen** [comment]
We haven't run the linter for minard-ui in Circle so far. If we decide to change that, I'd do that in another PR.

**Comment by dennari** [comment]
Ok, but the ones that it catches currently could be fixed here

**Comment by vsaarinen** [comment]
Changing the "no-lambda-in-jsx" warning isn't completely trivial, and is unrelated to this PR's content. I'd do it in a separate PR when needed.

**Pull request merged**